cpp/merge_tools/apply_differential_update.cc

```diff
@@ -92,23 +113,26 @@ void PatchArchiveMembersAndCreateOutputArchive(LocalDataDB * const local_data_db
     if (difference_archive_members.empty())
         LOG_WARNING("no difference archive members!");
 
+    std::unordered_set<std::string> title_ppns_in_delta;
+    std::unordered_set<std::string> authority_ppns_in_delta;
+    std::unordered_set<std::string> authority_ppns_in_input;
+
     //
     // We process title data first and combine all inferior and superior records.
     //
 
     const auto title_writer(MARC::Writer::Factory(output_directory + "/tit.mrc", MARC::FileType::BINARY));
-    std::unordered_set<std::string> previously_seen_title_ppns;
     CopySelectedTypes(local_data_db, difference_archive_members, title_writer.get(), { BSZUtil::TITLE_RECORDS, BSZUtil::SUPERIOR_TITLES },
-                      &previously_seen_title_ppns);
+                      &title_ppns_in_delta, authority_ppns_in_input, false);
     CopySelectedTypes(local_data_db, input_archive_members, title_writer.get(), { BSZUtil::TITLE_RECORDS, BSZUtil::SUPERIOR_TITLES },
-                      &previously_seen_title_ppns);
+                      &title_ppns_in_delta, authority_ppns_in_input, false);
 
     const auto authority_writer(MARC::Writer::Factory(output_directory + "/aut.mrc", MARC::FileType::BINARY));
-    std::unordered_set<std::string> previously_seen_authority_ppns;
+    PreFetchPPNSOfInput(input_archive_members, { BSZUtil::AUTHORITY_RECORDS }, &authority_ppns_in_input);
     CopySelectedTypes(local_data_db, difference_archive_members, authority_writer.get(), { BSZUtil::AUTHORITY_RECORDS },
-                      &previously_seen_authority_ppns);
+                      &authority_ppns_in_delta, authority_ppns_in_input, true);
     CopySelectedTypes(local_data_db, input_archive_members, authority_writer.get(), { BSZUtil::AUTHORITY_RECORDS },
-                      &previously_seen_authority_ppns);
+                      &authority_ppns_in_delta, authority_ppns_in_input, false);
```

Comment:
As soon as we pass authority_ppns_in_delta to the function a second time, it will no longer contain only delta ppn's, so we should find  a better name for this variable.

Reply:
... or => overload the function and only pass it in the first call